test/GPv2Settlement/Settle/Settle.t.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.26;
+
+import {Vm} from "forge-std/Test.sol";
+
+import {Harness, Helper} from "../Helper.sol";
+
+import {GPv2Settlement, GPv2Interaction} from "src/contracts/GPv2Settlement.sol";
+
+import {Settlement} from "test/libraries/Settlement.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+
+// solhint-disable func-name-mixedcase
+contract Settle is Helper {
+    using Settlement for Harness;
+    using SettlementEncoder for SettlementEncoder.State;
+
+    function setUp() public override {
+        super.setUp();
+    }
+
+    function test_allowlist_rejects_transactions_from_non_solvers() public {
+        vm.expectRevert("GPv2: not a solver");
+        settlement.settle(encoder.encode(settlement));
+    }
+
+    function test_allowlist_accepts_transactions_from_solvers() public {
+        vm.prank(solver);
+        settlement.settle(encoder.encode(settlement));
+    }
+
+    function test_emits_a_settlement_event() public {
+        vm.expectEmit(address(settlement));
+        emit GPv2Settlement.Settlement(solver);
+        vm.prank(solver);
+        settlement.settle(encoder.encode(settlement));
+    }
+
+    function test_executes_interactions_stages_in_the_correct_order() public {
```

Comment:
It doesn't look like this test is doing what it says. We don't verify the order of the events, and anyway we insert the interaction in the expected order, which wouldn't catch a misconfiguration of the way we insert interactions into the encoder.

Reply:
Will remove these tests in favour of E2E tests later.